code/game/objects/structures.dm

```diff
@@ -240,13 +240,17 @@
 
 //Damage
 /obj/structure/proc/take_damage(dam)
-	if(!dam || CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE|UNACIDABLE))
+	if(!dam || CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE))
 		return
 
 	obj_integrity = max(0, obj_integrity - dam)
 
 	if(obj_integrity <= 0)
-		playsound(src, 'sound/effects/metal_crash.ogg', 35)
-		qdel(src)
+		structure_destruction()
 	else
 		update_icon()
+
+
+/obj/structure/proc/structure_destruction()
+	playsound(src, 'sound/effects/metal_crash.ogg', 35)
```

Comment:
Don't make your own proc, if you really want this implement the proper /tg/ solution of object deconstruciton.

Reply:
@Rohesie, if you didn't intend this i can work to implement it with the other objects?

Reply:
What I'm asking is either do this another way or port the huge object damage rework from /tg/, not something half-baked.

Reply:
This still needs changing. This is not the proper way to do it and shouldn't be done in this PR.